### PR TITLE
Add language to block bad candidates.

### LINKED ISSUE #2531: Describe what happens with administratively prohibited candidates
If a TCP candidate is connecting to a port on the Do-not-connect list (from Fetch), something should happen to the candidate, and a connection attempt should not be made.

It seems OK to do what Chrome currently does, and just ignore the candidate - it does not show up in stats, in remote-description or in GetRemoteCandidates.

### LINKED ISSUE #2533: Do not allow DNS candidates in "relay" mode
This article:
https://www.digitaltrends.com/news/signal-vulnerability-hack-location/
described how to use DNS-valued candidates to get a rough geolocation capability.
This doesn't seem sensible to guard against when normal WebRTC procedures are in use, since the attacker can cause a call setup attempt to servers under his control too, but seems to be relevant when "relay" mode is in use, since that mode was explicitly introduced to avoid revealing the user's IP address-based location.

Suggested fix: Don't use this type of candidate when policy is "relay".

https://w3c.github.io/webrtc-pc/#dom-rtcicetransportpolicy-relay

Add:

"In this mode, remote candidates requiring resolution via some external mechanism, such as FQDN-valued candidates (require DNS lookup) and .local candidates (require MDN lookup), will be discarded."

### LINKED ISSUE #2534: Allow adminstrative rejection of candidates
In conjunction with the portscanning issue, it was stated (https://github.com/w3c/webrtc-pc/issues/2426#issuecomment-570528505) that "I think we should state that ICE-TCP ports should not connect to ports on the Fetch "bad ports" list - https://fetch.spec.whatwg.org/#port-blocking"

There might be other reasons for administratively rejecting a candidate - for instance overuse of resources, or local policy.

This seems to deserve its own issue.

### PR DESCRIPTION
**Pull request by alvestrand**
Fixes #2534
Fixes #2533 
Fixes #2531

***
<a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-pc/pull/2540.html" title="Last updated on Jul 9, 2020, 2:50 PM UTC (7bf1051)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/webrtc-pc/2540/30651fd...7bf1051.html" title="Last updated on Jul 9, 2020, 2:50 PM UTC (7bf1051)">Diff</a>

### CODE REVIEW

File: webrtc.html [hunk 1]
```diff
@@ -3153,6 +3155,25 @@ <h2>Methods</h2>
                     </ol>
                   </li>
                 </ol>
+                <p>
+                  A candidate is <dfn>administratively prohibited</dfn> if the
+                  UA has decided not to allow connection attempts to this
+                  address.
```
Comment (alvestrand) [inline webrtc.html:3161]: Sentence break here; the definition ends here.

File: webrtc.html [hunk 2]
```diff
@@ -3153,6 +3155,25 @@ <h2>Methods</h2>
...
+                <p>
+                  If the {{RTCConfiguration/iceTransportPolicy}} member of the {{RTCConfiguration}} is {{RTCIceTransportPolicy/relay}}, candidates requiring external
+                  resolution, such as mDNS candidates and DNS candidates,
+                  MUST be prohibited.
+                </p>
+                <p>
+                  For privacy
+                  reasons, there is no indication to the user about whether
+                  or not an address/port is blocked; it behaves exactly as
+                  if there was no response from the address.
```
Comment (alvestrand) [inline webrtc.html:3175]: Move this para to definition.

File: webrtc.html [hunk 3]
```diff
@@ -3153,6 +3155,25 @@ <h2>Methods</h2>
...
+                  address.
+                  The UA MUST prohibit connections to addresses on the
+                  [[!Fetch]] [= block bad port =] list, and MAY choose
+                  to not make connections to other addresses.
+                </p>
+                <p>
+                  If the {{RTCConfiguration/iceTransportPolicy}} member of the {{RTCConfiguration}} is {{RTCIceTransportPolicy/relay}}, candidates requiring external
+                  resolution, such as mDNS candidates and DNS candidates,
+                  MUST be prohibited.
+                </p>
```
Comment (alvestrand) [inline webrtc.html:3170]: Consider rejecting these candidates rather than silently dropping them.

File: webrtc.html (hunk 2 above)
Comment (dontcallmedom) [reply webrtc.html:3175]: I wouldn't use "user" here (which would typically be used for the end-user, not the developer); Fetch uses "developer", so I guess we could re-use that language here.

Comment (fippo) [reply webrtc.html:3175]: the developer might still be able to figure this out based on lack of candidate pair stats (which is what bit us re tcp) or lack of remote stats?

Comment (alvestrand) [reply webrtc.html:3175]: It's OK to let the developer learn whether or not there was a TURN server that accepted his credentials, or a peer that accepted his ufrag/pw, at that address.
What's not OK is to let the developer know the difference between a blocked port and a port where nobody's home. I think that's what this is trying to say.

**Pull request merged**